### PR TITLE
Quick takes

### PR DESCRIPTION
**Pull request by oetherington**
This branch adds the redesign of shortform -> quick takes.

Note that there's also a manual migration that needs to be run to update the titles of shortform posts (I've already done this on dev but not staging or prod).

After this PR any reference to "shortforms" on the EA forum is a bug, but I probably have missed some.

[Figma](https://www.figma.com/file/d09iVzNSpZp9GyWGcfkWmX/Q2.2-2023?type=design&node-id=13-13200&t=ol2rAOgXf6tizHy4-0)

Screenshot 2023-06-14 at 14 57 18

Screenshot 2023-06-14 at 14 57 29

Screenshot 2023-06-14 at 14 57 43

Screenshot 2023-06-14 at 14 58 47

Screenshot 2023-06-14 at 14 58 10

Screenshot 2023-06-14 at 14 58 04

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1204778335706771) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by Raemon** [comment]
I am curious about what this'll turn out to be

**Comment by oetherington** [comment]
@agnesstenlund I still need to tidy up the code in a couple of places before code review, but this is probably ready for you to take a first look at now.

**Comment by oetherington** [comment]
For now, I've not implemented turning the comment icon bold for new responses and this is pretty tricky - I think it's worth separating out into a separate PR.

**Comment by agnesstenlund** [comment]
This is beautiful 👏  Made some small padding + font adjustments.

A few things functionality wise – (may find some more as I keep looking 🤓)

- [ ] I'm not getting any feedback when I hit publish, and the text stays inside of the input (maybe this was the case before, too 😱). The input should be emptied and if we can't show the quick take straight away in the list, it would be nice to show some sort of small toast in the bottom right saying it was published 🤔 


https://github.com/ForumMagnum/ForumMagnum/assets/13235378/103658f4-3aa5-43ae-9548-e9a0490486a3


- [ ] Would be great to still use the comment-style font when you're writing the shortform too (to make it feel more informal and look more like the end result? Right now it looks more like a post)

Screenshot 2023-06-16 at 14 42 44
Screenshot 2023-06-16 at 14 48 10

- [ ] Hiding the author icon from the quick take itself would also be nice, since it's obvious this is the author (I think this is a left-over from the comments UI?)
Screenshot 2023-06-16 at 14 53 10

- [ ] I can still comment on my own quick takes page (although not on others, which is great). When I do so I get an error:

Screenshot 2023-06-16 at 15 05 01
perfect solution would be to show the quick takes input here instead

**Comment by agnesstenlund** [comment]
Small UI thing: even nicer if the `Publish` button has the same hover style as the one we use when you publish a post:

https://github.com/ForumMagnum/ForumMagnum/assets/13235378/7d31d1e7-7b51-4b7f-b720-eb8762664de7

**Comment by agnesstenlund** [comment]
also a little bit unexpected to see this symbol here when I click on the input field, but not big deal if it's hard to remove
Screenshot 2023-06-16 at 16 02 06

**Comment by agnesstenlund** [comment]
oh also seeing this in context I think we should be using the filled in icon for shortform, to better match the rest
Screenshot 2023-06-16 at 15 02 03

chat-bubble-left

File: packages/lesswrong/lib/ckEditorUtils.ts [hunk 1]
```diff
@@ -19,6 +31,8 @@ export function generateTokenRequest(collectionName: CollectionNameString, field
           return reject(new Error('Cannot download a new token!'));
         }
 
+        cache.set(cacheKey, xhrResponse);
```
Comment (Will-Howard) [inline packages/lesswrong/lib/ckEditorUtils.ts:34]: Is this specifically related to the quick takes changes or was this a problem before?

File: packages/lesswrong/server/manualMigrations/2023-06-13-renameShortformToQuicktakes.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  idempotent: true,
+  action: async () => {
+    await getSqlClientOrThrow().none(`
+      UPDATE "Posts"
+      SET title = (
+        SELECT "displayName"
+        FROM "Users"
+        WHERE "Posts"."userId" = "Users"."_id"
+      ) || '''s Quick takes'
+      WHERE "shortform" IS TRUE
```
Comment (Will-Howard) [inline packages/lesswrong/server/manualMigrations/2023-06-13-renameShortformToQuicktakes.ts:16]: Is there a reason this isn't an automatic migration?

**Review by Will-Howard — COMMENTED** [review]
A few miscellaneous things:
 - I think the comment box should shrink again if you click out of it when there is no text in it
 - It would be great if collapsing the "quick takes" section kept the comment box contents when you reopened it again (I can see people idling toggling the section and then losing all their text)
 - The dropdown item in the core topic page still says shortform
 - I'm sure Cmd-Enter used to submit a shortform on the frontpage but now it doesn't

File: packages/lesswrong/components/editor/Editor.tsx [hunk 3]
```diff
@@ -263,7 +272,7 @@ export const isBlank = (editorContents: EditorContents): boolean => {
   
   if (editorContents.type === "draftJS") {
     const draftJScontent = editorContents.value.getCurrentContent()
-    return !draftJScontent.hasText();
+    return draftJScontent.hasText();
```
[the remark this answers is not in the document]
Comment (oetherington) [reply packages/lesswrong/components/editor/Editor.tsx:275]: This is an accidental change 😬 - nicely caught!

Yes, we still support draftjs

File: packages/lesswrong/server/manualMigrations/2023-06-13-renameShortformToQuicktakes.ts (hunk 2 above)
Comment (oetherington) [reply packages/lesswrong/server/manualMigrations/2023-06-13-renameShortformToQuicktakes.ts:16]: Only that it makes merge conflicts a lot less likely 😂

File: packages/lesswrong/components/quickTakes/QuickTakesListItem.tsx [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+  const primaryTag = quickTake.relevantTags?.[0];
+  const displayHoverOver = hover && (quickTake.baseScore > -5) && !isMobile();
+
+  const {
+    LWTooltip, LWPopper, KarmaDisplay, UsersName, FooterTag, CommentsMenu,
+    ForumIcon, CommentsNode,
+  } = Components;
+
+
+  if (expanded) {
```
[the remark this answers is not in the document]
Comment (oetherington) [reply packages/lesswrong/components/quickTakes/QuickTakesListItem.tsx:123]: wtf

**Review by Will-Howard — APPROVED** [review]
ASTLAC (assuming it's just fixing the tooltip things that's left)

**Comment by oetherington** [comment]
@agnesstenlund This is probably ready for you to take another look at now. I think I've addressed everything except:

- I couldn't reproduce the problem where you weren't getting feedback after publishing - when I publish it clears the text entry and adds the new quick take to the list below (albeit, a little slowly). Is this still happening for you or not?
- Have you got any opinions on Will's point "I think the comment box should shrink again if you click out of it when there is no text in it"? FWIW, I think it's potentially tricky because of some edge cases, and if it goes wrong then it risks loosing the partially-written quick take, so I'm tempted to forget about it for now and then to make a separate PR if we want it later on.

Thanks!

**Comment by agnesstenlund** [comment]
>  Is this still happening for you or not?
It still is 😢 

https://github.com/ForumMagnum/ForumMagnum/assets/13235378/c48687b3-7d91-4db4-aeb5-0a9217f95cff



>  I'm tempted to forget about it for now and then to make a separate PR if we want it later on.

 I agree with this. (I also agree with Wills observation about the ideal behaviour, but don't think it's a dealbreaker)

**Comment by agnesstenlund** [comment]
Two small things about creating a quick take in the modal:

- [ ] 1. Would be great to automatically focus the input field when you click "New Quick take"

- [ ] 2. It's a bit wide–is that a component limitation or could we cut it down to around 750px?

Screenshot 2023-06-22 at 12 37 15

**Comment by agnesstenlund** [comment]
- [ ] something weird happened to the author line-break when I added a bunch of topics :)

Screenshot 2023-06-22 at 12 44 26

**Comment by agnesstenlund** [comment]
also all the fixes look and work really great 😍

**Comment by agnesstenlund** [comment]
I get an error when I try to comment, but assuming that's just a localhost thing rather than a real error?

https://github.com/ForumMagnum/ForumMagnum/assets/13235378/5ce4ca7b-8ea8-409a-ba21-ddf46ae7de13

**Comment by agnesstenlund** [comment]
- [ ] I think something weird happens when I click on the comment icon and on the copy link icon–for me neither of them do anything, or at least not what I would expect? Comment icon would ideally expand comments inline, but if that's tricky I'd expect them to open the comments up on that persons quick take page, which I don't think they do? As for the copy link icon I don't know if anything happens either, I don't get sent anywhere at least. And it doesn't get copied to clipboard

- [ ] this may be hard to do, but it'd be nice if the whole cell was clickable (rather than just the body text, which also have a hover). So the ideal thing would be: whole cell is clickable to expand it, but only when hovering the body text do we show the hover-over thing

**Comment by oetherington** [comment]
@agnesstenlund I think it should all be working now 🤞 - sorry about the back-and-forth!

It turns out that the bugs were happening because your account on the dev database wasn't approved. After speaking to JP, we decided to just hide the text input for unapproved users for now and then we can fix the problems and reenable it in a future PR. I also approved your dev account.

Let me know if you find anything else! Thanks!

**Comment by agnesstenlund** [comment]
dang weird, it's still not working for me 😮 

https://github.com/ForumMagnum/ForumMagnum/assets/13235378/231edaa8-4a52-4ede-bb5a-7d96aa86c66c

**Pull request merged**